pkg/app/pipedv1/plugin/kubernetes/deployment/primary_test.go

```diff
@@ -0,0 +1,197 @@
+	service, err := dynamicClient.Resource(serviceRes).Namespace("default").Get(ctx, "simple", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get service: %v", err)
+	}
+	assert.Equal(t, "simple", service.GetName())
+}
+
+func TestPlugin_executeK8sPrimaryRolloutStage_withPrune(t *testing.T) {
+	t.Parallel()
+
+	ctx := t.Context()
+
+	// initialize tool registry
+	testRegistry := toolregistrytest.NewTestToolRegistry(t)
+
+	// initialize deploy target config and dynamic client for assertions with envtest
+	dtConfig, dynamicClient := setupTestDeployTargetConfigAndDynamicClient(t)
+
+	runningOk := t.Run("prepare running state", func(t *testing.T) {
+		running := filepath.Join("testdata", "prune", "running")
+		runningCfg := sdk.LoadApplicationConfigForTest[kubeConfigPkg.KubernetesApplicationSpec](t, filepath.Join(running, "app.pipecd.yaml"), "kubernetes")
+
+		runningInput := &sdk.ExecuteStageInput[kubeConfigPkg.KubernetesApplicationSpec]{
+			Request: sdk.ExecuteStageRequest[kubeConfigPkg.KubernetesApplicationSpec]{
+				StageName:   "K8S_PRIMARY_ROLLOUT",
+				StageConfig: []byte(`{"prune":true}`),
+				TargetDeploymentSource: sdk.DeploymentSource[kubeConfigPkg.KubernetesApplicationSpec]{
+					ApplicationDirectory:      running,
+					CommitHash:                "0123456789",
+					ApplicationConfig:         runningCfg,
+					ApplicationConfigFilename: "app.pipecd.yaml",
+				},
+				Deployment: sdk.Deployment{
+					PipedID:       "piped-id",
+					ApplicationID: "app-id",
+				},
+			},
+			Client: sdk.NewClient(nil, "kubernetes", "", "", logpersistertest.NewTestLogPersister(t), testRegistry),
+			Logger: zaptest.NewLogger(t),
+		}
+
+		plugin := &Plugin{}
+		status := plugin.executeK8sPrimaryRolloutStage(ctx, runningInput, []*sdk.DeployTarget[kubeConfigPkg.KubernetesDeployTargetConfig]{
+			{
+				Name:   "default",
+				Config: *dtConfig,
+			},
+		})
+		assert.Equal(t, sdk.StageStatusSuccess, status)
+
+		deploymentRes := schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "deployments"}
+		_, err := dynamicClient.Resource(deploymentRes).Namespace("default").Get(ctx, "simple", metav1.GetOptions{})
+		assert.NoError(t, err)
+		serviceRes := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "services"}
+		_, err = dynamicClient.Resource(serviceRes).Namespace("default").Get(ctx, "simple", metav1.GetOptions{})
+		assert.NoError(t, err)
+	})
+	require.True(t, runningOk, "prepare running state subtest failed, aborting")
+
+	t.Run("prune with target state", func(t *testing.T) {
+		target := filepath.Join("testdata", "prune", "target")
+		targetCfg := sdk.LoadApplicationConfigForTest[kubeConfigPkg.KubernetesApplicationSpec](t, filepath.Join(target, "app.pipecd.yaml"), "kubernetes")
+
+		running := filepath.Join("testdata", "prune", "running")
+		runningCfg := sdk.LoadApplicationConfigForTest[kubeConfigPkg.KubernetesApplicationSpec](t, filepath.Join(running, "app.pipecd.yaml"), "kubernetes")
```

Comment:
[ask] Is it worthwhile to separate the test data?
The actual app.pipecd.yaml doesn't have the pipeline. But this stage is used only for pipeline sync.

Reply:
I agree it is confusing if we use configuration without the pipeline in the pipeline stage test cases.
I'll separate the test data.

Reply:
I added the test data for K8S_PRIMARY_ROLLOUT https://github.com/pipe-cd/pipecd/pull/5797/commits/cc8b5d8edecea27e679ec11104984356f9692ff7
And, switched to use it https://github.com/pipe-cd/pipecd/pull/5797/commits/61e25ee9127dc5a1091dddea4b655611423cb019

Reply:
Oops, test data's description is wrong. I'll fix it.

Reply:
I fixed testdata's name and description.
https://github.com/pipe-cd/pipecd/pull/5797/commits/c9575e2fcc9f4c05507203e11661685af81b30fe